addons/JobChange/jobchange.lua

```diff
@@ -0,0 +1,218 @@
+--[[
+Copyright © 2017, Sammeh of Quetzalcoatl
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions are met:
+
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of JobChange nor the
+      names of its contributors may be used to endorse or promote products
+      derived from this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL Sammeh BE LIABLE FOR ANY
+DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+(INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+]]
+
+_addon.name = 'Job Change'
+_addon.author = 'Sammeh'
+_addon.version = '1.0.2'
+_addon.command = 'jc'
+
+-- 1.0.1 first release
+-- 1.0.2 added 'reset' command to simply reset to existing job.  Changes sub job to a random starting job and back.
+
+require('tables')
+packets = require('packets')
+res = require ('resources')
+
+function jobchange(job,main_sub)
+    windower.add_to_chat(4,"JobChange: Changing "..main_sub.." job to:"..res.jobs[job].ens)
+    if job and main_sub then 
+        if main_sub == 'main' then 
+            local packet = packets.new('outgoing', 0x100, {
+            ["Main Job"]=job,
+            ["Sub Job"]=0,
+            ["_unknown1"]=0,
+            ["_unknown2"]=0
+            })
+            packets.inject(packet)
+            coroutine.sleep(0.5)
+        elseif main_sub == 'sub' then
+            local packet = packets.new('outgoing', 0x100, {
+            ["Main Job"]=0,
+            ["Sub Job"]=job,
+            ["_unknown1"]=0,
+            ["_unknown2"]=0
+            })
+            packets.inject(packet)
+            coroutine.sleep(0.5)
+        end
+    end
+end
+
+windower.register_event('addon command', function(command, ...)
+    local args = L{...}
+    local job = ''
+    if args[1] then 
+        job = args[1]:lower()
+    end
+    local currentjob = windower.ffxi.get_player()
```

Comment:
You call the expensive `windower.ffxi.get_player` function, yet you never use the result. However, below you call the same expensive function several times again, instead of using this value. I would pass this value to `find_conflict` and `find_temp_job` to avoid expensive recomputations of the same value.